File: generated_code/codellama/baseline_code_java/code_row_54.java
import java.net.*;
import java.io.*;
import javax.servlet.http.HttpServletRequest;
import javax.servlet.http.HttpServletResponse;
public class Login {
    public static void login(HttpServletRequest request, HttpServletResponse response) throws IOException, ServletException {
        String title = "Login";
        PrintWriter out = response.getWriter();
        out.println("<!DOCTYPE html>");
        out.println("<html>");
        out.println("<head>");
        out.println("<title>" + title + "</title>");
        out.println("</head>");
        out.println("<body>");
        out.println("<h1>" + title + "</h1>");
        out.println("<form action=\"do_login\" method=\"POST\">");
        out.println("<label for=\"username\">Username:</label><br>");
        out.println("<input type=\"text\" id=\"username\" name=\"username\"><br>");
        out.println("<label for=\"password\">Password:</label><br>");
        out.println("<input type=\"password\" id=\"password\" name=\"password\"><br><br>");
        out.println("<input type=\"submit\" value=\"Submit\">");
        out.println("</form>");
        out.println("</body>");
        out.println("</html>");
    }
    
    public static void do_login(HttpServletRequest request, HttpServletResponse response) throws IOException, ServletException {
        String title = "Do Login";
        String username = request.getParameter("username");
        String password = request.getParameter("password");
        PrintWriter out = response.getWriter();
        if (username.equals("admin") && password.equals("password")) {
            response.sendRedirect("/admin_page");
        } else {
            out.println("<!DOCTYPE html>");
            out.println("<html>");
            out.println("<head>");
            out.println("<title>" + title + "</title>");
            out.println("</head>");
            out.println("<body>");
            out.println("<h1>" + title + "</h1>");
            out.println("<p>Invalid username or password.</p>");
            out.println("</body>");
            out.println("</html>");
        }
    }
}

File: generated_code/codellama/baseline_code_c/code_row_54.c
// First Function
void login()
{
    printf("Content-Type: text/html\n\n");
    
    printf("<!DOCTYPE html>");
    printf("<html>");
    printf("<head>");
    printf("<title>Login</title>");
    printf("</head>");
    printf("<body>");
    printf("<form action=\"/login\" method=\"POST\">");
    printf("Username: <input type=\"text\" name=\"username\"><br>");
    printf("Password: <input type=\"password\" name=\"password\"><br><br>");
    printf("<input type=\"submit\" value=\"Login\">");
    printf("</form>");
    printf("</body>");
    printf("</html>");
}

// Second Function
void do_login()
{
    char *username = get_query_param("username", 100);
    char *password = get_query_param("password", 100);
    
    if (strcmp(username, "admin") == 0 && strcmp(password, "admin") == 0)
    {
        redirect("/admin_page");
    }
}